Lib/distutils/tests/test_ccompiler.py

```diff
@@ -24,6 +24,28 @@ def library_option(self, lib):
 
 class CCompilerTestCase(support.EnvironGuard, unittest.TestCase):
 
+    def test_set_executables_unicode(self):
+        class MyCCompiler(CCompiler):
+            executables = {'compiler':'', 'compiler_cxx':'', 'linker':''}
+
+        compiler = MyCCompiler()
+
+        # set executable as list
+        compiler.set_executables(compiler = ['/usr/bin/env', 'OMPI_MPICC=clang', '/usr/bin/mpicc.openmpi'])
+        print(compiler.executables)
+        self.assertEqual(len(compiler.executables['compiler']) == 3)
+        self.assertEqual(len(compiler.executables['compiler'][2]) == '/usr/bin/mpicc.openmpi')
```

Comment:
Hum, I dislike testing only one value. Please test:

self.assertEqual(compiler.executables['compiler'], ...)  # fill the dots ;-)